src/sentry/static/sentry/app/components/splitDiff.jsx

```diff
@@ -0,0 +1,106 @@
+import React, {PropTypes} from 'react';
+import classNames from 'classnames';
+import {diffChars, diffWords, diffLines} from 'diff';
+
+import '../../less/components/splitDiff.less';
+
+const diffFnMap = {
+  chars: diffChars,
+  words: diffWords,
+  lines: diffLines
+};
+
+const SplitDiff = React.createClass({
+  propTypes: {
+    base: PropTypes.string,
+    target: PropTypes.string,
+    type: PropTypes.oneOf(['lines', 'words', 'chars'])
+  },
+
+  getDefaultProps() {
+    return {
+      type: 'lines'
+    };
+  },
+
+  getInitialState() {
+    return {};
+  },
+
+  render() {
+    let {className, type, base, target} = this.props;
+    let cx = classNames('split-diff', className);
+    let diffFn = diffFnMap[type];
+
+    if (typeof diffFn !== 'function') return null;
+
+    let baseByLines = base.split('\n');
```

Comment:
Nitpick: I find `baseLines` clearer if this is just an array. `baseByLines` suggests this is an object/dictionary to me (e.g. "base keyed by lines").